doc/user_guide/user_guide.md

```diff
@@ -1,4 +1,152 @@
 # User Guide
 
+This project contains a python library for describing Exasol error messages. 
+This library lets you define errors with a uniform set of attributes. 
+Furthermore, the error message  is implemented to be parseable, 
+so that you can extract an error catalog from the code.
 
-This user guide provides you with usage examples for this repository. 
\ No newline at end of file
+## Attributes of Error Builder
+
+Error messages are built in `ExaError` using following predefined attributes. 
+All attributes of the error object except the error code are optional. Thus, 
+after the error-code is provided, whether other attributes are provided or not 
+will not cause any error. Please keep in mind that error-code should satisfy 
+error-code format (see [error-code](#error-code)).
+
+Flexibility is provided by introducing placeholder parameters to the error 
+message in two ways: (1) via the `parameter` attribute and (2) with positional 
+arguments. Furthermore, placeholders without parameters are filled with the 
+<null>.
+
+#### error-code
+This attribute should be defined in the error message and provide the following 
+pattern (`^[FEW]-[[A-Z][A-Z0-9]+(-[A-Z][A-Z0-9]+)*-[0-9]+$`):
+
+`severity "-" project-short-tag ["-" module-short-tag] "-" error-number` where,
+- _severity_: either F (Failure, not recoverable), or E (Error, recoverable), 
+or W (warning),
+- _project-short-tag_: alphanumeric starting with alphabet.
+- _module-short-tag_: alphanumeric starting with alphabet.
+- _error-number_: only number.
+
+Examples of valide error codes:
+- E-EXA-22004
+- E-EXA-SQL-22004
+- F-VS-QRW-13
+
+#### message
+This attribute includes error description which can be given by either as static 
+string or containing placeholders in double curly brackets. Parameters of 
```

Comment:
"by either as static 
string or containing placeholders in double curly brackets"
->
"by either a static 
string or a string containing placeholders in double curly brackets"